### DISCUSSION TITLE
Integration tests with pytest fixtures

### DISCUSSION THREAD

**Discussion by michalzan** [Neoteroi/BlackSheep#429] [Q&A]
I am creating an integration test with pytest fixture and following the official documentation. I have copy-pasted the fixtures from there, but I have a problem, that the API fixture is not of the Application type, but AsyncGenerator type, so as expected it does not work. Am I missing something? Here is an image for better understanding:
<img width="712" alt="Screenshot 2023-10-30 at 11 28 14" src="https://github.com/Neoteroi/BlackSheep/assets/28907686/c6a45445-a46e-4242-86d7-3774c0f6bece">
Thanks for all the comments and advice.

**Comment by tyzhnenko** [comment]
Hi @michalzan 

Here's my fixture, it works well. 

```python

@pytest.fixture
async def api_client(mongodb, init_db) -> TestClient:
    from tmw_server.app import app

    await app.start()

    return AuthTestClient(app)
```

Did you add to pytest.ini `asyncio_mode=auto`?

```ini
[pytest]
asyncio_mode=auto
```
